lib/marginalia.rb

```diff
@@ -112,6 +112,14 @@ def record_query_comment
     end
   end
 
+  def self.with_comment_settings(settings)
+    prev = settings.keys.map { |key| [key, Marginalia::Comment.send(key)] }
+    settings.each { |key, value| Marginalia::Comment.send(:"#{key}=", value) }
+    yield
+  ensure
+    prev.each { |(key, value)| Marginalia::Comment.send(:"#{key}=", value) }
+  end
+
```

Comment:
Appears to be a test helper. Would move this to a utility method on the test case rather than introduce new toplevel API.